query.go

```diff
@@ -314,8 +211,36 @@ checkResponses:
 			// stuck. This is a caveat for callers that should be
 			// fixed before exposing this function for public use.
 			checkResponse(sm.sp, sm.msg, quit)
+
+		// The current peer we're querying has failed to answer the
+		// query. Time to select a new peer and query it.
+		case <-peerTimeout:
+			select {
+			case peerQuit <- struct{}{}:
+			default:
+			}
+
+			curPeer = nil
+			for _, curPeer = range s.Peers() {
+				if curPeer != nil && curPeer.Connected() &&
```

Comment:
Perhaps we should try to _always_ not use the peer we used prior? As if the peer had issues just a bit ago, likely maybe it still won't be able to respond properly.

Reply:
Is that different from just setting the number of retries to 1?

Reply:
Hmm, I guess not.